### PR TITLE
Add cloud-first release notes for v23.2.12

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes REL-1358

### CODE REVIEW

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  - The `node_shutdown_transaction_timeout` event is logged after the timeout defined by the [cluster setting `server.shutdown.transactions.timeout` transpires]({% link v23.2/cluster-settings.md %}), if there are still open [transactions]({% link v23.2/transactions.md %}) on those SQL connections. [#128710][#128710]
+- Added the `ranges.decommissioning` metric, representing the number of [ranges]({% link v23.2/architecture/overview.md %}#architecture-range) which have a [replica]({% link v23.2/architecture/overview.md %}#architecture-replica) on a [decommissioning node]({% link v23.2/cockroach-node.md %}#decommission-nodes). [#130413][#130413]
+- Added three new network tracking metrics:
+  - `rpc.connection.connected` is the number of rRPC TCP level connections established to remote nodes.
+  - `rpc.client.bytes.egress` is the number of TCP bytes sent via gRPC on connections we initiated.
+  - `rpc.client.bytes.ingress` is the number of TCP bytes received via gRPC on connections we initiated. [#130712][#130712]
+- Added a new configuration parameter `server.cidr_mapping_url` which maps IPv4 CIDR blocks to arbitrary tag names. [#130712][#130712]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.12.md:27]:
```suggestion
- Added a new configuration parameter `server.cidr_mapping_url`, which maps IPv4 CIDR blocks to arbitrary tag names. [#130712][#130712]
```

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  - [Webhook sinks]({% link v23.2/changefeed-sinks.md %}#webhook-sink). If the [`server.child_metrics.enabled`]({% link v23.2/cluster-settings.md %}#setting-server-child-metrics-enabled) cluster setting is enabled, the metric will have a `webhook` label.
+  - [Pub/Sub sinks]({% link v23.2/changefeed-sinks.md %}#google-cloud-pub-sub). If the [`server.child_metrics.enabled`]({% link v23.2/cluster-settings.md %}#setting-server-child-metrics-enabled) cluster setting is enabled, the metric will have a `pubsub` label.
+  - [SQL sink]({% link v23.2/changefeed-for.md %}). If the [`server.child_metrics.enabled`]({% link v23.2/cluster-settings.md %}#setting-server-child-metrics-enabled) cluster setting is enabled, the metric will have a `sql` label. [#130712][#130712]
+
+<h3 id="v23-2-12-db-console-changes">DB Console changes</h3>
+
+- The metrics graphs in [DB Console]({% link v23.2/ui-overview-dashboard.md %}) now have hover behavior that focuses on individual lines and shows values under the mouse pointer. [cockroachdb/cockroach#128864][#128864]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.12.md:37]:
```suggestion
- The [DB Console]({% link v23.2/ui-overview-dashboard.md %}) time-series graphs now have hover behavior that focuses on individual lines and shows values under the mouse pointer. [cockroachdb/cockroach#128864][#128864]
```

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+- Users with the [`VIEWACTIVITY` privilege]({% link v23.2/security-reference/authorization.md %}#viewactivity) can download [statement bundles]({% link v23.2/explain-analyze.md %}) from [DB Console]({% link v23.2/ui-overview.md %}). [#129502][#129502]
+- Show an alert message in the [DB Console]({% link v23.2/ui-overview.md %}) when the [license is expired]({% link v23.2/licensing-faqs.md %}#monitor-for-license-expiry) or if there is less than 15 days left before it expires. [#130509][#130509]
+- [DB Console]({% link v23.2/ui-overview.md %}) will show a notification alerting customers without an [Enterprise license]({% link v23.2/enterprise-licensing.md %}) to [upcoming license changes](https://www.cockroachlabs.com/enterprise-license-update/) with a link to more information. [#130509][#130509]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.12.md:40]:
```suggestion
- The [DB Console]({% link v23.2/ui-overview.md %}) will now show a notification alerting customers without an [Enterprise license]({% link v23.2/enterprise-licensing.md %}) to [upcoming license changes](https://www.cockroachlabs.com/enterprise-license-update/) with a link to more information. [#130509][#130509]
```

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  - [SQL sink]({% link v23.2/changefeed-for.md %}). If the [`server.child_metrics.enabled`]({% link v23.2/cluster-settings.md %}#setting-server-child-metrics-enabled) cluster setting is enabled, the metric will have a `sql` label. [#130712][#130712]
+
+<h3 id="v23-2-12-db-console-changes">DB Console changes</h3>
+
+- The metrics graphs in [DB Console]({% link v23.2/ui-overview-dashboard.md %}) now have hover behavior that focuses on individual lines and shows values under the mouse pointer. [cockroachdb/cockroach#128864][#128864]
+- Users with the [`VIEWACTIVITY` privilege]({% link v23.2/security-reference/authorization.md %}#viewactivity) can download [statement bundles]({% link v23.2/explain-analyze.md %}) from [DB Console]({% link v23.2/ui-overview.md %}). [#129502][#129502]
+- Show an alert message in the [DB Console]({% link v23.2/ui-overview.md %}) when the [license is expired]({% link v23.2/licensing-faqs.md %}#monitor-for-license-expiry) or if there is less than 15 days left before it expires. [#130509][#130509]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.12.md:39]:
```suggestion
- [DB Console]({% link v23.2/ui-overview.md %}) now displays an alert message when the [license is expired]({% link v23.2/licensing-faqs.md %}#monitor-for-license-expiry) or if there are fewer than 15 days left before the license expires. [#130509][#130509]
```

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+- [DB Console]({% link v23.2/ui-overview.md %}) will show a notification alerting customers without an [Enterprise license]({% link v23.2/enterprise-licensing.md %}) to [upcoming license changes](https://www.cockroachlabs.com/enterprise-license-update/) with a link to more information. [#130509][#130509]
+
+<h3 id="v23-2-12-bug-fixes">Bug fixes</h3>
+
+- Declarative and legacy [schema changes]({% link v23.2/online-schema-changes.md %}) were incorrectly allowed to be executed concurrently before, which could lead to failing or hung schema change jobs. [#128838][#128838]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.12.md:44]:
```suggestion
- Fixed a bug where declarative and legacy [schema changes]({% link v23.2/online-schema-changes.md %}) were incorrectly allowed to be executed concurrently, which could lead to failing or hung schema change jobs. [#128838][#128838]
```

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+- Fixed a bug that caused errors like `ERROR: column 'crdb_internal_idx_expr' does not exist` when accessing a table with an [expression index]({% link v23.2/expression-indexes.md %}) where the expression evaluates to an [`ENUM`]({% link v23.2/enum.md %}) type, e.g., `CREATE INDEX ON t ((col::an_enum))`. [#129092][#129092]
+- [Function]({% link v23.2/user-defined-functions.md %}) input parameters can no longer be of the `VOID` type. [#129281][#129281]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.12.md:46]:
```suggestion
- [Function]({% link v23.2/user-defined-functions.md %}) input parameters can no longer have the `VOID` type. [#129281][#129281]
```

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+- Internally issued queries that are not initiated within a [SQL session]({% link v23.2/show-sessions.md %}) no longer respect a statement timeout. This includes: [background jobs]({% link v23.2/show-jobs.md %}), queries issued by the [DB Console]({% link v23.2/ui-overview.md %}) that perform introspection, and the Cloud SQL shell. [#129517][#129517]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.12.md:47]:
```suggestion
- Internally issued queries that are not initiated within a [SQL session]({% link v23.2/show-sessions.md %}) no longer respect a statement timeout. This includes: [background jobs]({% link v23.2/show-jobs.md %}), queries issued by the [DB Console]({% link v23.2/ui-overview.md %}) that perform introspection, and the {{ site.data.products.cloud }} [SQL shell]({% link cockroachcloud/sql-shell.md %}). [#129517][#129517]
```

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+- Fixed a bug where the `schema_locked` [storage parameter]({% link v23.2/with-storage-parameter.md %}) did not prevent a table from being referenced by a [foreign key]({% link v23.2/foreign-key.md %}). [#129753][#129753]
+- Users with the [`VIEWACTIVITY` SQL privilege]({% link v23.2/security-reference/authorization.md %}#viewactivity) can now request, view and cancel [statements bundles]({% link v23.2/explain-analyze.md %}) in the [DB Console]({% link v23.2/ui-overview.md %}). [#129803][#129803]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.12.md:49]:
```suggestion
- Users with the [`VIEWACTIVITY` SQL privilege]({% link v23.2/security-reference/authorization.md %}#viewactivity) can now request, view, and cancel [statement bundles]({% link v23.2/explain-analyze.md %}) in the [DB Console]({% link v23.2/ui-overview.md %}). [#129803][#129803]
```

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+- Fixed a bug where a [lease transfer]({% link v23.2/architecture/replication-layer.md %}#epoch-based-leases-table-data) could lead to a panic with the message `side-transport update saw closed timestamp regression`. The bug could occur when a node was overloaded and failing to [heartbeat its node liveness record]({% link v23.2/cluster-setup-troubleshooting.md %}#node-liveness-issues). [#129808][#129808]
+- Resolved a concerning [log message]({% link v23.2/logging-overview.md %}) that said `expiration of liveness record ... is not greater than expiration of the previous lease ... after liveness heartbeat`. This message is no longer possible. [#129808][#129808]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.12.md:51]: We've published this note with this wording -- it's not clear to me what "resolved" means

```suggestion
- The [log message]({% link v23.2/logging-overview.md %}) `expiration of liveness record ... is not greater than expiration of the previous lease ... after liveness heartbeat` is no longer generated. [#129808][#129808]
```

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+- Fixed a bug where the `require_explicit_primary_keys` [session variable]({% link v23.2/set-vars.md %}) would prevent all [`CREATE TABLE`]({% link v23.2/create-table.md %}) statements from working. [#129906][#129906]
+- Fixed a slow-building memory leak when using [Kerberos authentication]({% link v23.2/gssapi_authentication.md %}). [#130317][#130317]
+- Fixed a potential memory leak in [changefeeds using a cloud storage sink]({% link v23.2/changefeed-examples.md %}#create-a-changefeed-connected-to-a-cloud-storage-sink). The memory leak could occur if both [`changefeed.fast_gzip.enabled`]({% link v23.2/cluster-settings.md %}#setting-changefeed-fast-gzip-enabled) and `changefeed.cloudstorage.async_flush.enabled` were true, and the changefeed received an error while attempting to write to the cloud storage sink. [#130624][#130624]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.12.md:54]:
```suggestion
- Fixed a potential memory leak in [changefeeds using a cloud storage sink]({% link v23.2/changefeed-examples.md %}#create-a-changefeed-connected-to-a-cloud-storage-sink). The memory leak could occur if both [`changefeed.fast_gzip.enabled`]({% link v23.2/cluster-settings.md %}#setting-changefeed-fast-gzip-enabled) and `changefeed.cloudstorage.async_flush.enabled` were `true`, and the changefeed received an error while attempting to write to the cloud storage sink. [#130624][#130624]
```

**Review by taroface — APPROVED** [review]
LGTM just buncha nits

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+<h3 id="v23-2-12-sql-language-changes">SQL language changes</h3>
+
+- The [session setting]({% link v23.2/set-vars.md %}) `plan_cache_mode=force_generic_plan` can now be used to force prepared statements to use a query plan that is [optimized]({% link v23.2/cost-based-optimizer.md %}) once and reused in future executions without re-optimization, as long as the plan does not become stale due to [schema changes]({% link v23.2/online-schema-changes.md %}) or a collection of new [table statistics]({% link v23.2/show-statistics.md %}). The setting takes effect during [`EXECUTE`]({% link v23.2/sql-grammar.md %}#execute_stmt) commands. [`EXPLAIN ANALYZE`]({% link v23.2/explain-analyze.md %}) includes a `plan type` field. If a generic query plan is optimized for the current execution, the `plan type` will be `generic, re-optimized`. If a generic query plan is reused for the current execution without performing optimization, the `plan type` will be `generic, reused`. Otherwise, the `plan type` will be `custom`. [#128100][#128100]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:14]: Updated all instances of `set-vars.md` to use this other page - although idk why we added another page without removing the old one (set-vars is the OG page).  anyway changed 'em all 

also snagged the edit - thanks!

✅

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+- The [session setting]({% link v23.2/set-vars.md %}) `plan_cache_mode=auto` can now be used to instruct the [cost-based optimizer]({% link v23.2/cost-based-optimizer.md %}) to automatically determine whether to use "custom" or "generic" query plans for the execution of a prepared statement. Custom query plans are optimized on every execution, while generic plans are optimized once and reused on future executions as-is. Generic query plans are beneficial in cases where query optimization contributes significant overhead to the total cost of executing a query. [#128100][#128100]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:15]: ✅

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+
+<h3 id="v23-2-12-operational-changes">Operational changes</h3>
+
+- There are now structured logging events that report connection breakage during node shutdown. Previously, these logs existed but were unstructured. These logs appear in the `OPS` [logging channel]({% link v23.2/logging-overview.md %}#logging-channels). There are two new events:
+  - The `node_shutdown_connection_timeout` event is logged after the timeout defined by the [cluster setting `server.shutdown.connections.timeout`]({% link v23.2/cluster-settings.md %}) transpires, if there are still open SQL connections.
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:20]: ✅

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+  - The `node_shutdown_transaction_timeout` event is logged after the timeout defined by the [cluster setting `server.shutdown.transactions.timeout` transpires]({% link v23.2/cluster-settings.md %}), if there are still open [transactions]({% link v23.2/transactions.md %}) on those SQL connections. [#128710][#128710]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:21]: ✅

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+- Added the `ranges.decommissioning` metric, representing the number of [ranges]({% link v23.2/architecture/overview.md %}#architecture-range) which have a [replica]({% link v23.2/architecture/overview.md %}#architecture-replica) on a [decommissioning node]({% link v23.2/cockroach-node.md %}#decommission-nodes). [#130413][#130413]
+- Added three new network tracking metrics:
+  - `rpc.connection.connected` is the number of rRPC TCP level connections established to remote nodes.
+  - `rpc.client.bytes.egress` is the number of TCP bytes sent via gRPC on connections we initiated.
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:25]: pinged engineer for clarity on this usage of 'we' (see your DMs)

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+  - `rpc.client.bytes.ingress` is the number of TCP bytes received via gRPC on connections we initiated. [#130712][#130712]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:26]: see above re: pinged engineer for clarification

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+  - The `node_shutdown_connection_timeout` event is logged after the timeout defined by the [cluster setting `server.shutdown.connections.timeout`]({% link v23.2/cluster-settings.md %}) transpires, if there are still open SQL connections.
+  - The `node_shutdown_transaction_timeout` event is logged after the timeout defined by the [cluster setting `server.shutdown.transactions.timeout` transpires]({% link v23.2/cluster-settings.md %}), if there are still open [transactions]({% link v23.2/transactions.md %}) on those SQL connections. [#128710][#128710]
+- Added the `ranges.decommissioning` metric, representing the number of [ranges]({% link v23.2/architecture/overview.md %}#architecture-range) which have a [replica]({% link v23.2/architecture/overview.md %}#architecture-replica) on a [decommissioning node]({% link v23.2/cockroach-node.md %}#decommission-nodes). [#130413][#130413]
+- Added three new network tracking metrics:
+  - `rpc.connection.connected` is the number of rRPC TCP level connections established to remote nodes.
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:24]: ✅

File: src/current/_includes/releases/v23.2/v23.2.12.md (hunk 1 above)
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:27]: ✅

File: src/current/_includes/releases/v23.2/v23.2.12.md (hunk 2 above)
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:37]: ✅

File: src/current/_includes/releases/v23.2/v23.2.12.md (hunk 3 above)
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:40]: ✅

File: src/current/_includes/releases/v23.2/v23.2.12.md (hunk 4 above)
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:39]: ✅

File: src/current/_includes/releases/v23.2/v23.2.12.md (hunk 5 above)
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:44]: ✅

File: src/current/_includes/releases/v23.2/v23.2.12.md (hunk 6 above)
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:46]: ✅

File: src/current/_includes/releases/v23.2/v23.2.12.md (hunk 7 above)
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:47]: ✅

File: src/current/_includes/releases/v23.2/v23.2.12.md (hunk 8 above)
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:49]: ✅

File: src/current/_includes/releases/v23.2/v23.2.12.md (hunk 9 above)
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:51]: ✅

File: src/current/_includes/releases/v23.2/v23.2.12.md (hunk 10 above)
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:54]: ✅

File: src/current/_includes/releases/v23.2/v23.2.12.md [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+- The [session setting]({% link v23.2/set-vars.md %}) `plan_cache_mode=auto` can now be used to instruct the [cost-based optimizer]({% link v23.2/cost-based-optimizer.md %}) to automatically determine whether to use "custom" or "generic" query plans for the execution of a prepared statement. Custom query plans are optimized on every execution, while generic plans are optimized once and reused on future executions as-is. Generic query plans are beneficial in cases where query optimization contributes significant overhead to the total cost of executing a query. [#128100][#128100]
+
+<h3 id="v23-2-12-operational-changes">Operational changes</h3>
+
+- There are now structured logging events that report connection breakage during node shutdown. Previously, these logs existed but were unstructured. These logs appear in the `OPS` [logging channel]({% link v23.2/logging-overview.md %}#logging-channels). There are two new events:
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.12.md:19]: ✅

**Comment by rmloveland** [comment]
ok still haven't heard back from engineer - not gonna hold the train for this minor edit, if it becomes super important we can always come back and update 

(also i wanna merge this during biz hours since there may be some wonkiness to work thru w.r.t. cloud 2.0 merge and the inevitable merge conflict in releases.yml)

**Pull request merged**